sale_packaging_default/models/sale_order_line.py

```diff
@@ -43,6 +46,23 @@ def _compute_product_packaging_id(self):
                     and line.product_uom_qty % line.product_packaging_id.qty
                 ):
                     line.product_packaging_id = False
+                    res = line._warning_product_packaging_required()
+                    if res:
+                        return res
+        return result
+
+    @api.depends("product_id", "product_uom_qty", "product_uom")
+    def _warning_product_packaging_required(self):
+        """Notify the user when a packaging is required."""
+        result = {}
+        for line in self:
+            if line.require_packaging and not line.product_packaging_id:
+                raise UserError(
+                    _(
+                        "The product %s requires a packaging to be sold"
+                        % line.product_id.name
+                    )
+                )
         return result
```

Comment:
issue: use a constrain instead, and I summarize all the comments from above here too:

```suggestion

    @api.constrains("product_id", "product_packaging_id")
    def _check_product_packaging_required(self):
        """Notify the user when a packaging is required."""
        for line in self:
            if line.require_packaging and not line.product_packaging_id:
                raise ValidationError(
                    _(
                        "The product %s requires a packaging to be sold",
                        line.product_id.name
                    )
                )
```